waku/waku_rln_relay/rln_relay.nim

```diff
@@ -435,19 +434,20 @@ proc mount(
   (await groupManager.startGroupSync()).isOkOr:
     return err("could not start the group sync: " & $error)
 
+  if (conf.rlnRelayUserMessageLimit > 20):
```

Comment:
it should be 100 now, as per parameters decided by the waku research team :)

Reply:
Thanks for the review, @rymnc. I will updates in the next commit.

Reply:
mm, why enforcing this limit? the user is free to choose whatever `rlnRelayUserMessageLimit` they want. TheWakuNetwork sets this to a max of 100, but it shouldn't be enforced in the code.

for example, someone may want to deploy their own cluster, with their own contract with rlnRelayUserMessageLimit=9999.

Reply:
Hi @AlejandroCabeza, Alvaro makes a good point about why we are enforcing a limit for messages.

Reply:
> mm, why enforcing this limit? the user is free to choose whatever `rlnRelayUserMessageLimit` they want. TheWakuNetwork sets this to a max of 100, but it shouldn't be enforced in the code.
> 
> for example, someone may want to deploy their own cluster, with their own contract with rlnRelayUserMessageLimit=9999.

agree - this limit should probably be enforced in another place -
`OnchainGroupManager`: fetch the `MAX_MESSAGE_LIMIT` from the contract and compare with the user provided `rlnRelayUserMessageLimit`

Reply:
Great ! then let's automate that.

Reply:
Hi @rymnc, I tried fetching MAX_MESSAGE_LIMIT from the contract based on your suggestion, but it failed in many cases. Upon debugging, I discovered that it returns 0 because it hasn't been initialized yet. Now, I've found another approach: there's an attribute called userMessageLimit in base_groupmanager, which I believe is a better option. What do you think ?

Earlier, as you mentioned that the message limit is set 20 for static. Can we fetch the on-chain limit from the contract?

Reply:
> Can we fetch the on-chain limit from the contract?

that is possible by calling `MAX_MESSAGE_LIMIT()`

Reply:
> I discovered that it returns 0

please share a code snippet?

Reply:
I've just updated the PR. plz take a look at.

cc @rymnc

Reply:
The associated test case failed with this implementation because the MAX_MESSAGE_LIMIT() API is returning 0. This is likely due to it not being initialized, which is why it returns zero.